tensorflow/contrib/layers/python/layers/layers.py

```diff
@@ -1481,6 +1483,57 @@ def max_pool2d(inputs,
 
 
 @add_arg_scope
+def maxout(inputs,
+           num_units,
+           axis=None,
+           outputs_collections=None,
+           scope=None):
+  """Adds a maxout op from https://arxiv.org/abs/1302.4389
+
+    "Maxout Networks"
+
+    Ian J. Goodfellow, David Warde-Farley, Mehdi Mirza, Aaron Courville, Yoshua
+    Bengio
+
+
+  Max pooling is performed in given filter/channel dimension. This can also be
+  used after fully-connected layers to reduce number of features.
+
+
+  Args:
+    inputs: A Tensor on which maxout will be performed
+    num_units: Specifies how many features will remain after max pooling at the
+      channel dimension. This must be multiple of number of channels.
+    axis: The dimension where max pooling will be performed. Default is the
+      last dimension.
+    outputs_collections: The collections to which the outputs are added.
+    scope: Optional scope for name_scope.
+
+  Returns:
+    A `Tensor` representing the results of the pooling operation.
+
+  Raises:
+    ValueError: if num_units is not multiple of number of features.
+    """
+  with ops.name_scope(scope, 'MaxOut', [inputs]) as sc:
+    inputs = ops.convert_to_tensor(inputs)
+    shape = inputs.get_shape().as_list()
+    if axis is None:
+      # Assume that channel is the last dimension
+      axis = -1
+    num_channels = shape[axis]
+    if num_channels % num_units:
+      raise ValueError('number of features({}) is not '
+                       'a multiple of num_units({})'
+              .format(num_channels, num_units))
+    shape[axis] = -1
```

Comment:
should it be `= num_units`?
[it's better not to have -1 in reshape, as it's easy to wrongly reshape and not know about it]